### ISSUE TITLE
False positive, unavoidable incosistent indentation

### ISSUE THREAD

**Issue by lolmaus** [clutchski/coffeelint#389]
Here's my piece of CoffeeScript:

```coffee
  selectedPeopleCount: Ember.computed \
    'allPeopleChecked',
    'selectedPeople.length',
    'totalUsers',
    ->
      if @get 'allPeopleChecked'
        return @get 'totalUsers'

      @get 'selectedPeople.length'
```

It raises an error:

> #48: Line contains inconsistent indentation. Expected 2 got 4.

The line in question is the `if` one.

The funny thing is that it's impossible to write this snippet so that CoffeeLint stops complaining. If i unindent the `if` line, then CoffeeLint throws the same error for the next line, the `return` one. Obviously, this also breaks the code.

I've got no idea how to refactor this code to satisfy CoffeeLint. It only stops the false positive error if i write all arguments in one line, but then it throws a "line exceeds max length" error.

**Comment by AsaAyers** [comment]
```CoffeeScript
  selectedPeopleCount: Ember.computed \
      'allPeopleChecked', # double indent
      'selectedPeople.length', # double indent
      'totalUsers', # double indent
  -> # don't indent
    # ^ Doing the above gets this if to be indented 1 level deeper than `selectedPeopleCount`
    if @get 'allPeopleChecked'
      return @get 'totalUsers'
    @get 'selectedPeople.length'
```

For more info see the comments at #384

**Comment by lolmaus** [comment]
It passses CoffeeLint this way, but do you believe that this is a proper way to write code? To my eye, absolutely unnatural.

**Comment by AsaAyers** [comment]
Honestly, I recommend [Babel](http://babeljs.io/). In my other projects I've started moving away from CoffeeScript and using ES6 instead.

You're welcome to see if you can figure out a way to get CoffeeLint to accept the code the way you want to write it. I'm just saying that I haven't figured out a way so this is where we are. I joined this project because it didn't work the way I wanted it to, I wanted to fix things and add some missing functionality.

**Comment by lolmaus** [comment]
@AsaAyers, i do use Babel in my project, but i don't see how it can replace CoffeeScript.

As for the matter discussed in this issue, here's another piece of valid CoffeeScript rejected by CoffeeLint:

```coffee
foo = bar.baz \
  quux,
  zomg: ->
    lol()
```

At the same time, this works:

```coffee
foo = bar.baz  quux,
  zomg: ->
    lol()
```

But this might not be possible because then the first line may exceed max length.

To fix this according to @AsaAyers suggestion, this code should be refactored as:

```coffee
foo = bar.baz \
    quux,
  zomg: ->
    lol()
```

Which makes it unreadable. You have to make extra effort to decipher ridiculous indentation imposed by CoffeeLint.

> You're welcome to see if you can figure out a way to get CoffeeLint to accept the code the way you want to write it.

The first thing to do here is to acknowledge that CoffeeLint has a problem by reopening this issue and assigning a `bug` label.

Currently, introducing CoffeeLint to my project just makes the codebase worse.

**Comment by AsaAyers** [comment]
> @AsaAyers, i do use Babel in my project, but i don't see how it can replace CoffeeScript.

Sorry, I meant that it replaces CoffeeScript by writing ES6 instead of writing in CoffeeScript. Then you can use linters that have indentation rules more suited to what you're trying to do.

Upon your recommendation I have reopened #384 because it has better explanations of what's going on. I've even created a `Help Wanted` label for it.

**Comment by lolmaus** [comment]
@AsaAyers, you've only added tags. Please reopen. ^_^

**Comment by lolmaus** [comment]
My teammate @gleb-osokin suggested a more readable indentation fix:

```coffee
foo = bar.baz \
  quux,
  huh,
    zomg: ->
      lol()
    love: 'hate'
```

Readable enough for me! No crazy unindented lines, no double indentation, obvious code structure.

**Comment by AsaAyers** [comment]
I'm all for supporting a more natural syntax. Just need someone to get over
the technical hurdles to get CoffeeLint to accept it.

**Comment by swang** [comment]
@lolmaus, alot of these issues with indentation is due to CoffeeScript rewriting tokens at the lexer level. CoffeeLint is forced then to sometimes not rely on its tokens info, but rather manually evaluate each individual line. 

At this point it may be easier to petition CoffeeScript's developers to show us the true raw tokens or delay rewrites until after the lexer stage. They closed the last issue because this issue is resolved in CoffeeScriptRedux, but I don't ever see CoffeeScriptRedux becoming the mainline CoffeeScript implementation. Even if it did, we'd probably have to rewrite CoffeeLint since I don't believe they expose the token data anymore.

Also, from this:
```cs
foo = bar.baz \
  quux,
  huh,
    zomg: ->
      lol()
    love: 'hate'
```
I just realized that CS allows backslashes to slurp up the next line even if its not a string. Wonder if that's purposely done or by accident.

**Comment by lolmaus** [comment]
CoffeeScriptRedux is abandonned. It hasn't seen active development for half a year, stuck at 666 commits (sic).

**Comment by swang** [comment]
I agree it most likely is which is why I think a new issue should be opened up in the CS repo to take another look at the problem.

**Comment by lolmaus** [comment]
I don't argue that, but CS maintainer is a reactionary, still refusing to implement ES6-compliant imports. I doubt that he will refactor CS internals to our liking.
